### PR TITLE
feat: add bootcamps url restructuring method to command

### PR DESCRIPTION
**Pull request by AfaqShuaib09**
This PR modifies `migrate_course_slugs` command to add bootcamps URL restructuring method to cmd

### CODE REVIEW

File: course_discovery/apps/course_metadata/constants.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 REFRESH_COURSE_SKILLS_URL_NAME = 'refresh_course_skills'
 REFRESH_PROGRAM_SKILLS_URL_NAME = 'refresh_program_skills'
 COURSE_UUID_REGEX = r'[0-9a-f-]+'
-SUBDIRECTORY_SLUG_FORMAT_REGEX = r'learn\/[a-zA-Z0-9-_]+\/[a-zA-Z0-9-_]+$|executive-education\/[a-zA-Z0-9-_]+$'
+SUBDIRECTORY_SLUG_FORMAT_REGEX = r"learn\/[a-zA-Z0-9-_]+\/[a-zA-Z0-9-_]+$|executive-education\/[a-zA-Z0-9-_]+$|boot-camps\/[a-zA-Z0-9-_]+\/[a-zA-Z0-9-_]+$"
```
Comment (AliAdnanSohail) [inline course_discovery/apps/course_metadata/constants.py]: nit: use use single quote `'` to be consistent with other regex variables

**Review by AliAdnanSohail — APPROVED** [review]
LGTM! Merge this when build is green

File: course_discovery/apps/course_metadata/utils.py [hunk 2]
```diff
@@ -1022,6 +1022,22 @@ def create_slug_for_exec_ed():
             slug = f"executive-education/{organization_slug}-{course_slug}"
         return slug, error
 
+    def create_slug_for_bootcamps():
```
Comment (Ali-D-Akbar) [inline course_discovery/apps/course_metadata/utils.py:1028]: docstring for this

**Review by Ali-D-Akbar — APPROVED** [review]

File: course_discovery/apps/course_metadata/tests/test_utils.py [hunk 3]
```diff
@@ -1082,6 +1082,28 @@ def test_get_slug_for_exec_ed_course(self):
...
+
+        slug, error = utils.get_slug_for_course(course)
+        subject = SubjectFactory(name='business')
+        course.subjects.add(subject)
+        slug, error = utils.get_slug_for_course(course)
+
+        assert error is None
+        assert slug == f"boot-camps/{subject.slug}/{org.name}-{slugify(course.title)}"
+
     def test_get_slug_for_course__with_no_url_slug(self):
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/tests/test_utils.py]: nit: please divide this test into separate unit tests. This test is currently checking both happy & bad scenarios.

File: course_discovery/apps/course_metadata/utils.py [hunk 4]
```diff
@@ -1022,7 +1025,29 @@ def create_slug_for_exec_ed():
...
+        course_subjects = course.subjects.all()
+        if not course_subjects:
+            error = f"Bootcamp with uuid {course.uuid} and title {course.title} does not have any subject"
+            logger.info(error)
+            return None, error
+        primary_subject_slug = course_subjects[0].slug
+
+        slug = f'boot-camps/{primary_subject_slug}/{organization_slug}-{course_slug}'
+        if is_existing_slug(slug, course):
+            logger.info(f"Slug '{slug}' already exists in DB, recreating slug by adding a number in course_title")
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/utils.py]: nit: add an identifier to signify that this error is coming from bootcamp

File: course_discovery/apps/course_metadata/tests/test_utils.py [hunk 5]
```diff
@@ -1170,6 +1192,47 @@ def test_get_slug_for_exec_ed_course__with_existing_url_slug(self):
...
+        # duplicate a new course with same title, subject and organization
+        course_3 = CourseFactory(title='test-title', type=bootcamp_type, partner=partner)
+        course_3.authoring_organizations.add(org)
+        course_3.subjects.add(subject)
+        course_3.save()
+        CourseUrlSlug.objects.filter(course=course_3).delete()
+        slug, error = utils.get_slug_for_course(course_3)
+        assert error is None
+        assert slug == f"boot-camps/{subject.slug}/{org.name}-{slugify(course_3.title)}-3"
+        course_3.set_active_url_slug(slug)
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/tests/test_utils.py]: nit: this can be added in a range loop. The only thing changing is index/count.

Comment (AfaqShuaib09) [reply course_discovery/apps/course_metadata/tests/test_utils.py]: done

File: course_discovery/apps/course_metadata/utils.py (hunk 4 above)
Comment (AfaqShuaib09) [reply course_discovery/apps/course_metadata/utils.py]: Added Bootcamp keyword in logs

File: course_discovery/apps/course_metadata/tests/test_utils.py (hunk 3 above)
Comment (AfaqShuaib09) [reply course_discovery/apps/course_metadata/tests/test_utils.py]: done

**Review by DawoudSheraz — APPROVED** [review]

**Pull request merged**